File: app/src/main/java/br/monteoliva/mynfc/Principal.java
package br.monteoliva.mynfc;

// imports da API do ANDROID
import android.support.v7.app.ActionBar;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.Toolbar;
import android.view.KeyEvent;
import android.widget.TextView;
import android.content.Intent;

// imports da API MonteolivaLIB
import org.androidannotations.annotations.AfterViews;
import org.androidannotations.annotations.Click;
import org.androidannotations.annotations.EActivity;
import org.androidannotations.annotations.ViewById;

import br.monteoliva.monteolivalib.nfc.NFCScanner;
import br.monteoliva.monteolivalib.nfc.OnNFCListener;

/**
 * Tela Principal
 * 
 * @author Claudio Monteoliva
 * @version 1.0
 *
 * @copyright 2019 Monteoliva Developer
 */
@EActivity(R.layout.principal)
public class Principal extends AppCompatActivity implements OnNFCListener {
    private NFCScanner scanner;

    @ViewById(R.id.txtId)
    protected TextView txtId;

    protected ActionBar actionBar;

    @ViewById(R.id.toolbar)
    protected Toolbar toolbar;

    @AfterViews
    protected void afterViews() {
        // toolbar
        setSupportActionBar(toolbar);
        actionBar = getSupportActionBar();
        if (actionBar != null) {
            actionBar.setTitle("");
            actionBar.setSubtitle("");
        }

        // seta o NFC Scanner
        scanner = new NFCScanner(this, getIntent());
        scanner.setOnNFCListener(this);
    }

    @Click(R.id.btnClear)
    protected void btnClear() {
        txtId.setText("");
    }

    /**
     * Metodo de ciclo de vida onResume
     */
    @Override
    protected void onResume() {
        super.onResume();

        // onResume do NFC Scanner
        scanner.onResume();
    }

	/**
	 * Metodo de onPause
	 */
    @Override
    protected void onPause() {
    	super.onPause();

        // onPause do NFC Scanner
        scanner.onPause();
	}

    /**
     * Metodo para a Leitura do NFC
     *
     * @param intent
     */
    @Override
    public void onNewIntent(Intent intent) { scanner.resolveIntent(intent); }

    @Override
    public void onTagRead(String nrTag) {
        // coloca os dados
        txtId.setText(txtId.getText().toString() + nrTag + "\n");
    }

    /**
     * Evento KeyDown
     */
    @Override
    public boolean onKeyDown(int keyCode, KeyEvent event) {
        if (keyCode == KeyEvent.KEYCODE_BACK) { finish(); return true; }
        else                                  { return super.onKeyDown(keyCode, event); }
    }
}

File: app/src/main/java/br/monteoliva/mynfc/Principal.kt
package br.monteoliva.mynfc

import android.content.Intent
import android.os.Bundle
import android.view.KeyEvent
import android.widget.Button
import android.widget.TextView

import androidx.appcompat.app.ActionBar
import androidx.appcompat.app.AppCompatActivity
import androidx.appcompat.widget.Toolbar

import br.monteoliva.monteolivalib.nfc.NFCScanner
import br.monteoliva.monteolivalib.nfc.OnNFCListener

/**
 * Tela Principal
 *
 * @author Claudio Monteoliva
 * @version 1.0
 *
 * @copyright 2020 Monteoliva Developer
 */
class Principal : AppCompatActivity(), OnNFCListener {
    private lateinit var scanner: NFCScanner
    private lateinit var txtId: TextView
    private lateinit var toolbar: Toolbar

    private var actionBar: ActionBar? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.principal)

        txtId = findViewById(R.id.txtId)
        toolbar = findViewById(R.id.toolbar)

        setSupportActionBar(toolbar)
        actionBar = supportActionBar
        if (actionBar != null) {
            actionBar!!.title = ""
            actionBar!!.subtitle = ""
        }

        // seta o NFC Scanner
        scanner = NFCScanner(this, intent)
        scanner.setOnNFCListener(this)

        val btnClear = findViewById<Button>(R.id.btnClear)
            btnClear.setOnClickListener { txtId.setText("") }
    }

    override fun onResume() {
        super.onResume()

        // onResume do NFC Scanner
        scanner.onResume()
    }

    override fun onPause() {
        super.onPause()

        // onPause do NFC Scanner
        scanner.onPause()
    }

    /**
     * Metodo para a Leitura do NFC
     *
     * @param intent
     */
    public override fun onNewIntent(intent: Intent) {
        super.onNewIntent(intent)
        scanner.resolveIntent(intent)
    }

    override fun onTagRead(nrTag: String) {
        txtId.text = "${txtId.text}$nrTag\n"
    }

    /**
     * Evento KeyDown
     */
    override fun onKeyDown(keyCode: Int, event: KeyEvent): Boolean {
        return if (keyCode == KeyEvent.KEYCODE_BACK) {
            finish()
            true
        } else {
            super.onKeyDown(keyCode, event)
        }
    }
}